vcd/datasource_vcd_catalog.go

```diff
@@ -141,34 +148,96 @@ func datasourceVcdCatalog() *schema.Resource {
 	}
 }
 
+func getOrgName(vcdClient *VCDClient, d *schema.ResourceData) string {
+	orgName := d.Get("org").(string)
+	if orgName == "" {
+		orgName = vcdClient.Org
+	}
+	return orgName
+}
+
+func getCatalogFromResource(catalogName string, d *schema.ResourceData, meta interface{}) (*govcd.AdminCatalog, error) {
+	vcdClient := meta.(*VCDClient)
+
+	orgName := getOrgName(vcdClient, d)
```

Comment:
Not sure if I'm missing something, but I think this could be simplified to
```suggestion
	orgName, err := vcdClient.GetOrgNameFromResource(d)
```
then `getOrgName` is no longer needed.